### PR TITLE
Fix bug 1353241 - 'View diff' for first revision doesn't work

### PR DESCRIPTION
**Pull request by harikishen**
Added a check for first data_version in releases.py

### CODE REVIEW

**Comment by harikishen** [comment]
@mozbhearsum do you want a test that checks for ISE 500 ?

**Comment by harikishen** [comment]
i was planning on adding a test in test_history.py which adds a new release and then requests the ViewDiff url for 200 status code ? What do you think ?

**Comment by mozbhearsum** [comment]
Yeah, a test that ensure viewing the diff for the first revision returns a 200.

**Comment by harikishen** [comment]
@mozbhearsum i've added  the test. Could you check it ?

File: auslib/test/admin/views/test_history.py [hunk 1]
```diff
@@ -96,6 +96,39 @@ def testFieldViewRelease(self):
...
+
+        ret = self._put('/releases/ddd1', data=dict(blob=blob, name='ddd1',
+                        product='d', data_version=1))
+        self.assertStatusCode(ret, 201)
+        ret = self._post(
+            '/releases/ddd1',
+            data=dict(data=blob, product='d', data_version=1)
+        )
+        self.assertStatusCode(ret, 200)
+
```
Comment (mozbhearsum) [inline auslib/test/admin/views/test_history.py]: This doesn't look like the test you want. The code you've changed is involved with GETs to endpoints like /api/history/diff/release/10189589/data - so you should be making requests like that in your test.

Comment (harikishen) [inline auslib/test/admin/views/test_history.py]: Yea I think I pasted the wrong code and forgot to change it.

**Comment by harikishen** [comment]
@mozbhearsum hope this is right.

File: auslib/test/admin/views/test_history.py [hunk 2]
```diff
@@ -96,6 +96,41 @@ def testFieldViewRelease(self):
...
+                    }
+                }
+            }
+        }"""
+
+        ret = self._put('/releases/ddd1', data=dict(blob=blob, name='ddd1',
+                        product='d', data_version=1))
+        self.assertStatusCode(ret, 201)
+        table = dbo.releases.history
+        row, = table.select(order_by=[table.change_id.desc()], limit=1)
```
Comment (mozbhearsum) [inline auslib/test/admin/views/test_history.py]: I think you want .asc() here. Right now, this ends up selecting the last change, not the first one. r=me with that fixed.

Comment (harikishen) [inline auslib/test/admin/views/test_history.py]: ohk will do that

File: auslib/test/admin/views/test_history.py [hunk 3]
```diff
@@ -123,10 +123,12 @@ def testFieldViewDiffFirstRelease(self):
         ret = self._put('/releases/ddd1', data=dict(blob=blob, name='ddd1',
```
Comment (harikishen) [inline auslib/test/admin/views/test_history.py]: @mozbhearsum does'nt this ensure that this is a new release ?

**Comment by harikishen** [comment]
@mozbhearsum did the changes you requested r=bhearsum

**Comment by mozbhearsum** [comment]
Unfortunately, the test is now failing. Have you been running tests locally before pushing? You should do that to make sure you don't push code or tests that are broken.

**Comment by harikishen** [comment]
@mozbhearsum i did check it

**Comment by mozbhearsum** [comment]
That's strange - the test is failing on our continuous integration. You can see the log here: https://public-artifacts.taskcluster.net/GRGrYHYHQiGfranPb47GpQ/0/public/logs/live_backing.log

**Comment by harikishen** [comment]
@mozbhearsum please look into it now

**Comment by mozbhearsum** [comment]
This looks good, but you'll need to merge in upstream before I can merge the PR.

**Comment by harikishen** [comment]
@mozbhearsum done merging with upstream.

**Pull request merged**